### ISSUE TITLE
Add variant of stairs function to match that one of Matplotlib

### ISSUE THREAD

**Issue by pmc4** [JuliaPlots/Makie.jl#3591]
Currently, the `stairs` function of Makie has the parameter `step` that allows three different values: `:pre`, `:post` and `:center`. This can be seen in the docs: https://docs.makie.org/stable/reference/plots/stairs/

However, in Matplotlib Python's package this function behaves a bit differently. The endpoint edges of the stairs plot go to vertically to the bottom of the plot, like a mix of the `:pre` and `:post` steps all together. This can be seen in the following Matplotlib example: https://matplotlib.org/stable/gallery/lines_bars_and_markers/stairs_demo.html

The other difference is that in [Makie's `stairs` ](https://docs.makie.org/stable/reference/plots/stairs/)the `xs` and `ys` values have the same number of elements. On the other hand, the [Matplotlib's one](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.stairs.html#matplotlib.pyplot.stairs) takes as `xs` the bin edges and as `ys` the central values. That is, the `xs` has one more element than `ys`.

### My suggestion
My suggestion is to create a variant of `stairs` that accepts as `xs` a vector of bin edges and as `ys` a vector of values for the mid points of each bin. Right now this can be done by computing the vertexes of the lines and using the `lines` function. I do so with this function:
```julia
"""
    stairs2lines(bin_edges, ys)

Convert an array of `bin_edges` and `y_axis_values` to two arrays (x, y) of the vertexes
needed to do the stair plot with a line plot.

Note that `length(bin_edges) = length(ys) + 1`.

Modify the `baseline` parameter to change the minimum value on the endpoint edges.
"""
function stairs2lines(bin_edges, y_axis_values; baseline = 0.0)
    xs = Float64[]
    ys = Float64[]

    total_bins = length(bin_edges)

    for i in eachindex(bin_edges)
        if i == 1
            x1 = bin_edges[i]
            x2 = bin_edges[i]
        elseif i == total_bins
            x1 = bin_edges[i]
            x2 = bin_edges[i]
        else
            x1 = bin_edges[i]
            x2 = bin_edges[i]
        end

        push!(xs, x1)
        push!(xs, x2)
    end

    for i in eachindex(bin_edges)
        if i == 1
            y1 = baseline
            y2 = y_axis_values[i]
        elseif i == total_bins
            y1 = y_axis_values[i-1]
            y2 = baseline
        else
            y1 = y_axis_values[i-1]
            y2 = y_axis_values[i]
        end

        push!(ys, y1)
        push!(ys, y2)
    end

    return xs, ys
end
```

An example of use is the following one:
```julia
f = Figure()
# Current Makie stairs
bin_edges = LinRange(0, 4pi, 21)
bin_midpoints = (xs[begin:end-1] + xs[begin+1:end])
ys = sin.(bin_midpoints)

stairs(f[1, 1], bin_midpoints, ys)
stairs(f[2, 1], bin_midpoints, ys; step=:post, color=:blue, linestyle=:dash)
stairs(f[3, 1], bin_midpoints, ys; step=:center, color=:red, linestyle=:dot)


# New variation of stairs
# Obtain the vertexes for the line plot
new_xs, new_ys = stairs2lines(bin_edges, ys; baseline = -1.0)
lines(f[4,1], new_xs, new_ys; color=:green)

f
```

Which produces the following figure
![fig](https://github.com/MakieOrg/Makie.jl/assets/117096890/db46e6d6-92cd-4178-be6d-3bcdfa3a1c3b)

First 3 subfigures are obtained using current `stairs` function and the fourth one is the variant. Note that in the first 3 I am plotting the bin midpoints and the computed values, whereas in the fourth I am plotting the bin edges and the same computed values. I use a `baseline = -1.0` to lower the lines to the bottom of the axis.

### Useful applications
In the field of high energy physics, it is common to find in papers that report the number of measured events in each energy bin. Basically you have an array of bin edges that delimit some bins. For each bin you have a number of events. With the variant version of stairs is very easy to do the plot. In Figure 14 of https://arxiv.org/pdf/2211.17120.pdf you can see an example.

Disclaimer: I chose that figure just for illustrative purposes. I am not affiliated with the LZ collaboration.

**Comment by jkrumbiegel** [comment]
Hm I'm not sure if I think this fits the API of `stairs`, because it's rather unusual that with a keyword argument, the structure of the positional arguments has to change. In this case, that the values have to be one less than the edges. Maybe this would rather warrant a new recipe then, the name of which code be `barlines` or something, to me what you want is more like a line trace version of a barplot. We already have `stephist` but this doesn't take the precomputed values.

Note also that you can get your desired plot by using the `:pre` step type, using the baseline for the first and last value, and adding the last point again to get the line down.

```julia
edges = 1:6
values = [3, 5, 2, 4, 1]
baseline = -3
stairs([edges; last(edges)], [baseline; values; baseline], step = :pre, axis = (; xticks = 1:6))
```
<img width="583" alt="grafik" src="https://github.com/MakieOrg/Makie.jl/assets/22495855/62691af1-a31f-4119-b33f-6fe503c51417">

**Comment by pmc4** [comment]
Yes, I agree that it should be a different recipe because it's weird to have with the same recipe a variation that needs `ys` to be 1 element shorter than `xs` and other variation that needs `ys` and `xs` of the same length.

Thanks for the alternative way of getting the same plot with stairs, though.

I think that a new recipe that either calls `stairs` with your version or `lines` with mine (or something similar) is desirable. Instead of having to transform your data, one can just send the x values, y values and the baseline, like in Matplotlib.
